### ISSUE TITLE
isDocker is not working on Kubernetes v1.23

### ISSUE THREAD

**Issue by QuanTran91** [fastify/fastify-cli#579]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.1.1

### Node.js version

16

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### Description

Plugin [is docker](https://github.com/sindresorhus/is-docker) is not work on EKS (Elastic Kubernetes Service) 1.23

### Steps to Reproduce

Run app in EKS as pod
Observe the log that it is running on 127.0.0.1

**Comment by climba03003** [comment]
I would say it is a upstream issue. and you are allowed to pass the address by yourself.

**Comment by sinedied** [comment]
I would say that isDocker works as expected since K8s doesn't implement a Docker engine, but it would be nice if the fastify-cli adds some detection logic to use 0.0.0.0 as default in K8s environments. I lost 2h yesterday because of this

**Comment by mcollina** [comment]
Ah! That's the catch. Do you know how can we detect that we running on K8s?

**Comment by sinedied** [comment]
Seems like you can check for an environment variable: https://stackoverflow.com/questions/36639062/how-do-i-tell-if-my-container-is-running-inside-a-kubernetes-cluster

**Comment by sinedied** [comment]
Would you accept a PR for this?

**Comment by climba03003** [comment]
> Would you accept a PR for this?

Yes, and I see more complete check on this package https://github.com/ntfs32/is-kubernetes/blob/main/index.js
The download is 1 / week, so I have a bit worry on the stability. But, the detection look solid to me.

**Comment by sinedied** [comment]
Ok, I'll propose a PR then. Though I would be against adding an external dependency for something that is a 1-liner, as the environment variable is enough (according to the docs, it's always present: https://kubernetes.io/docs/tutorials/services/connect-applications-service/#environment-variables).

**Comment by climba03003** [comment]
> Though I would be against adding an external dependency for something that is a 1-liner, as the environment variable is enough

I have no idea will this change in future but multiple method for detection doesn't hurt.

**Comment by sinedied** [comment]
Given that any change to this would break the entire k8s ecosystem, it's very unlikely.
I'd rather thrust the official doc than than an unknown repo using a detection methods based on synchronous file reads, which could delay the startup a bit. 

I've sent the PR BTW.

**Comment by climba03003** [comment]
> I'd rather thrust the official doc than than an unknown repo using a detection methods based on synchronous file reads, which could delay the startup a bit.

The check is `OR` which means it only execute when the first one fail. When the environment is always exist, there will be no file read operation. Again, fallback detection methods doesn't hurt in anyway.

When a environment really doesn't set one, then the application is no-way out either. Even worse then consuming sometime for reading file-system.

**Comment by Tony133** [comment]
This issue was resolved with PR [#580](https://github.com/fastify/fastify-cli/pull/580)
